### PR TITLE
Change AKTming

### PR DESCRIPTION
**Pull request by dave234**
This has a few breaking changes.

AKTiming adopters will have to rename a few functions. I don't believe anyone is adopting it at this point so probably no biggie.

AKClipRecoder.play was renamed to start. This might affect a few early adopters.

### CODE REVIEW

File: AudioKit/Common/Nodes/Playback/Player/ClipPlayer/AKClipPlayer.swift [hunk 1]
```diff
@@ -216,8 +186,36 @@ open class AKClipPlayer: AKNode, AKTiming {
         set { playerNode.pan = newValue }
     }
 }
-extension AKClipPlayer {
+extension AKClipPlayer: AKTiming {
```
Comment (dave234) [inline AudioKit/Common/Nodes/Playback/Player/ClipPlayer/AKClipPlayer.swift]: @ryanfrancesconi Moving the AKTiming functions to an extension makes it clear that the functions are there to satisfy AKTiming.  Maybe AKPlayer could do the same.

File: AudioKit/Common/Internals/Synchronization/AKTiming.swift [hunk 2]
```diff
@@ -12,33 +12,93 @@
...
-    func play(at audioTime: AVAudioTime?)
+    @objc func start(at audioTime: AVAudioTime?)
 
     /// Stops playback immediately.
-    func stop()
+    @objc func stop()
 
-    /// Start playback immediately.
-    func play()
+    var isStarted: Bool { get }
```
Comment (dave234) [inline AudioKit/Common/Internals/Synchronization/AKTiming.swift]: @ryanfrancesconi I removed play() from AKTiming it's just a convenience for play(at:nil) I'm sorry about isStarted, it had to be done.

Comment (ryanfrancesconi) [reply AudioKit/Common/Internals/Synchronization/AKTiming.swift]: ok cool. I just committed an AKPlayer update, so I'm going to look at this now

**Comment by dave234** [comment]
I think it’s alright.  The thing I like about setPosition and position(at audioTime) is that it’s not a convenience like var position set get, or currentTime set get.  This allows each implementor to have the conveniences as part of their class.  If someone is doing something AKTiming related, it should be with the intent to accurately schedule audio.  That was part of the reasoning behind removing the start() function.

**Comment by aure** [comment]
I think it will be wise to target the next release as 4.1 with this change and the new AKPlayer, and possibly a deprecation marking if not removal of AKAudioPlayer.

**Comment by ryanfrancesconi** [comment]
I would deprecate AKAudioPlayer as soon as some others have success with AKPlayer. I've been testing it in a few different scenarios, but there could be more?

**Comment by ryanfrancesconi** [comment]
I will implement this AKTiming, then I think it's ok to merge AKPlayer

**Pull request merged**